api/v1alpha1/applicationvolumereplication_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2021 The RamenDR authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FailoverClusterMap defines the clusters used for failover per subscription. Key is subscription name
+type FailoverClusterMap map[string]string
+
+// ApplicationVolumeReplicationSpec defines the desired state of ApplicationVolumeReplication
+type ApplicationVolumeReplicationSpec struct {
+	FailoverClusters FailoverClusterMap `json:"statuses,omitempty"`
```

Comment:
We will need the following (over time) as discussed,

- SubscriptionSelector: A selector to operate only on certain Subscriptions within the namespace. In the absence of which all Subscriptions in the namespace are acted upon.
    - Even if OCM restricts an application to a namespace, for OCM Subscriptions that are not bundled as an application in a namespace it would be useful to have a selector
    - This selector MAY/WOULD be different than the PVC Selector which gets applied on the managed clusters, for PVCs from channel sources. As the Subscription labels are Hub related and application channel resources may have their own labels
- ReplicationClass: To feed to VRG CR and subsequently feed to VR CRs
    - There are some gotcha's here, every PVC is assumed to belong to the same StorageClass and also ReplicationClass
- PVCSelector: To feed to VRG CR and subsequently feed to VR CRs
    - PVCSelector also allows to filter out PVCs that should not be DR protected (if any) within Subscriptions

Reply:
1. There's a proposal to generalize the name 'PVCSelector' to 'ReplicationResourceSelector' at https://github.com/RamenDR/ramen/pull/13

2. When both clusters are available (from ACM's availability point of view), what can the user do to force to the app to one of replicas, for reasons of his choice?

Reply:
I understand that AVR is one per app (add a comment about it please). Does the design of an AVR per subscription make it any simpler?

Reply:
I understand that you haven't considered failback design yet.  But, after setting FailoverClusters and failing over successfully, what is expected to happen if the user clears FailoverClusters?

Reply:
Consider the scenario where a subscription is active in MC1, which fails. ACM decides to place the app to the replicated MC2 and has paused the app waiting for user input to AVR.failedClusters. But the user wants to run the app on MC1 again (because MC1 is back and MC2 is out of date). What should the user do to AVR CR to ensure that ACM deploys the app on MC1? In the design without AVR, the user could achieve this goal. We need a way to do something similar in the ACM based AVR design.

Reply:
I believe our design needs to allow one subscription (say, prod) to be on a replicated storage with another subscription (say, test) on an unreplicated storage.

Reply:
FYI: Note that there's a proposal to generalize the name 'PVCSelector' to 'ReplicationResourceSelector' at https://github.com/RamenDR/ramen/pull/13

Reply:
The AVR is one per app, but the label selector that we will add will return only the subscriptions that match the label.

Reply:
@vdeenadh those are all good questions.  I have not spent the time thinking about the failback.  I am thinking that we should get the failover working and then tweak it for the fallback.